packages/viem-compatibility/src/compatibility.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  Log as EthersLog,
+  TransactionReceipt as EthersTransactionReceipt,
+} from '@ethersproject/abstract-provider'
+import { StaticJsonRpcProvider } from '@ethersproject/providers'
+import { BigNumber } from 'ethers'
+import {
+  Chain,
+  Client,
+  PublicClient,
+  Transport,
+  Log as ViemLog,
+  TransactionReceipt as ViemTransactionReceipt,
+} from 'viem'
+
+// based on https://wagmi.sh/react/ethers-adapters#reference-implementation
+export function publicClientToProvider<TChain extends Chain | undefined>(
```

Comment:
the orbit sdk added some extra stuff here: https://github.com/OffchainLabs/arbitrum-orbit-sdk/blob/main/src/ethers-compat/publicClientToProvider.ts

and there's also a test that we could copy over https://github.com/OffchainLabs/arbitrum-orbit-sdk/blob/main/src/ethers-compat/publicClientToProvider.unit.test.ts